js/pages/myToDoList.js

```diff
@@ -753,21 +753,32 @@ const setModuleAttributes = (data, modules, collections) => {
         modules['Covid-19'].enabled = true;
     }
 
-    if(data[fieldMapping.menstrualSurveyEligible] === 353358909) {
+    if (data[fieldMapping.menstrualSurveyEligible] === 353358909) {
         modules['Menstrual Cycle'].enabled = true;
 
-        if(data[fieldMapping.MenstrualCycle.statusFlag] !== fieldMapping.moduleStatus.submitted) {
+        if (data[fieldMapping.MenstrualCycle.statusFlag] !== fieldMapping.moduleStatus.submitted) {
 
             const cutoffDate = new Date();
-            cutoffDate.setDate(cutoffDate.getDate() - 45);
 
-            if(data[fieldMapping.Biospecimen.statusFlag] === fieldMapping.moduleStatus.submitted) {
-                if(data[fieldMapping.Biospecimen.completeTs] < cutoffDate.toISOString()) {
+            // Survey will become available on dashboard 120 hours (5 days) after completion of bio survey if participant is menstrual survey eligible
+            cutoffDate.setDate(cutoffDate.getDate() - 5);
```

Comment:
please avoid using the same date object if we are going to use it to set two different variables

Reply:
Solved